### ISSUE TITLE
Export withScaleable and useScaleable in order for external components to be written alongside Geist

### ISSUE THREAD

**Issue by sannajammeh** [geist-org/geist-ui#589]
# Feature request 🚀

  - [x] I will create Pull Request
  - [ ] It's just a suggestion

### Expected

  - The new scale function in Geist UI is an amazing new addition and contains the perfect way to extend custom UI components. It would be great if this was exported in the main entry alongside the rest of Geist UI. This is internal code, but it would make a fantastic addition to the library as external.  We're currently using this for a pet project with many custom components and Geist UI as our main UI library, the scale method is the most clever method to pixel-perfect optimize and build components while retaining the same Geist UI workflow in the project. 
  
 I will happily make a PR and documentation (if needed) for this.

### Examples

```jsx
import {useScaleable, withScaleable } from '@geist-ui/react/dist/use-scaleable';
// To
import  {useScaleable, withScaleable } from '@geist-ui/react';
```

**Comment by unix** [comment]
Exporting `useScaleable` and `withScaleable` is fine.
We need to add additional documentation to this hook to describe how it works, so these things may be necessary:

- Export `useScaleable` and `withScaleable` in `index.ts`
- Export the necessary types
- Add a document named `useScaleable`

If you would like to contribute these you can create PRs directly, if no one contributes after a while I will start this work.

**Comment by unix** [comment]
Implemented in version 2.3.0.
